File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/CollectContract.java
package com.android.learn.mcontract;

import com.android.learn.base.mmodel.FeedArticleListData;
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData;
import com.android.learn.base.mview.BaseView;

/**
 * Created by gaolei on 2018/6/18.
 */

public class CollectContract {

    public interface Presenter {


        void getCollectList();

        void cancelCollectArticle(int position, int id);
    }

    public interface View extends BaseView{

        void showCollectList(FeedArticleListData feedArticleListData);

        void showCancelCollectArticle(int position, int id);
    }
}

File: AndroidLearnProject/app/src/main/java/com/android/learn/mcontract/CollectContract.kt
package com.android.learn.mcontract

import com.android.learn.base.mmodel.FeedArticleListData
import com.android.learn.base.mmodel.FeedArticleListData.FeedArticleData
import com.android.learn.base.mview.BaseView

/**
 * Created by gaolei on 2018/6/18.
 */

class CollectContract {

    interface Presenter {


        fun getCollectList()

        fun cancelCollectArticle(position: Int, id: Int)
    }

    interface View : BaseView {

        fun showCollectList(feedArticleListData: FeedArticleListData)

        fun showCancelCollectArticle(position: Int, id: Int)
    }
}
